state/factory/factory.go

```diff
@@ -356,7 +360,8 @@ func (sf *factory) Validate(ctx context.Context, blk *block.Block) error {
 		if err := ws.ValidateBlock(ctx, blk); err != nil {
 			return errors.Wrap(err, "failed to validate block with workingset in factory")
 		}
-		sf.putIntoWorkingSets(key, ws)
+		sf.workingsets.Add(key, ws)
+		sf.workingsets.Add(ws.height, ws)
```

Comment:
this is success validation for `block n`, validation of the next block may continue which needs the state changes in this block, so adding the `ws` to cache, which will become parent of next block's workingset